Document tunnel option for development (#4163)
nextauthjs/next-auth: docs/docs/providers/apple.md

Apple doesn't allow you to use localhost in domains or subdomains.
:::

### Using a tunnel

> One developemnt option to avoid using HTTP or localhost is to access your site through a tunnel such as [ngrok](https://ngrok.com/).

Reviewer: ```suggestion
You can use a tunneling service like [ngrok](https://ngrok.com/) to simplify local development and easily get an HTTPS URL required for Apple OAuth settings.
```

> 1. Follow the instructions to set up the tunnel and install the agent.

Reviewer: ```suggestion
1. Follow the [instructions](https://ngrok.com/download) to set up an ngrok tunnel.
```

2. In `.env.local`, temporarily set `NEXTAUTH_URL` to be your assigned tunnel URL, something like `https://12345.ngrok.io`.
3. Temporarily add the tunnel domain (`12345.ngrok.io`) and reply URL (`https://12345.ngrok.io/api/auth/callback/apple`) into your sign-in service identifier in [Certificates, Identifiers & Profiles](https://developer.apple.com/account/resources/identifiers/serviceId/edit/). 
4. Browse to your tunnel URL and test sign in.

Once you've tested authentication and created a local account, you can remove the tunnel references and switch back to using localhost.

### Modifying your local host name

> An altnerative approach is to make your development computer see itself as a non-local website. Edit your host file and point your site to `127.0.0.1`.

Reviewer: ```suggestion
An alternative approach is to setup a hostname and SSL certificates for your local computer. 

Begin by editing your host file (`/etc/hosts`) and point your site to `127.0.0.1`. For example, by adding the line: `127.0.0.1 mysite.com`
```
